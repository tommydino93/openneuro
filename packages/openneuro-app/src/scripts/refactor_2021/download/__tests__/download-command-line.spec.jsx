import React from 'react'
import { shallow } from 'enzyme'
import { DownloadSampleCommand } from '../download-command-line.jsx'

const defProps = { datasetId: 'ds000001' }

describe('dataset/download', () => {
  describe('DownloadSampleCommand component', () => {
    it('renders successfully', () => {
      const wrapper = shallow(<DownloadSampleCommand {...defProps} />)
      expect(wrapper).toMatchSnapshot()
    })
    it('drafts show draft flag', () => {
      const wrapper = shallow(<DownloadSampleCommand {...defProps} />)
      expect(wrapper.text()).toEqual(expect.stringContaining('--draft'))
      expect(wrapper.text()).toEqual(expect.not.stringContaining('--snapshot'))
    })
    it('snapshots show snapshot flag', () => {
      const wrapper = shallow(
        <DownloadSampleCommand {...defProps} snapshotTag="1.0.0" />,
      )
      expect(wrapper.text()).toEqual(expect.stringContaining('--snapshot'))
      expect(wrapper.text()).toEqual(expect.not.stringContaining('--draft'))
    })
  })
})
