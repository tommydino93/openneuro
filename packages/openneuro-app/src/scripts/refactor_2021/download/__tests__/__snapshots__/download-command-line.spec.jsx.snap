// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/download DownloadSampleCommand component renders successfully 1`] = `
<Styled(pre)>
  openneuro download 
  --draft
   
  ds000001
   
  ds000001
  -download/
</Styled(pre)>
`;
